backend/src/main/java/com/redhat/cloud/notifications/routers/EndpointResource.java

```diff
@@ -162,7 +164,11 @@ public EndpointPage getEndpoints(
         for (Endpoint endpoint: endpoints) {
             // Fetch the secrets from Sources.
             if (this.featureFlipper.isSourcesUsedAsSecretsBackend()) {
-                this.secretUtils.loadSecretsForEndpoint(endpoint);
+                try {
+                    this.secretUtils.loadSecretsForEndpoint(endpoint);
+                } catch (final SourcesException e) {
+                    Log.error(e.getMessage(), e);
+                }
```

Comment:
I'll add comments to each case but it doesn't necessarily mean I'm requesting changes. I just want to make sure I understand each of them.

Reply:
In this case, we're loading multiple endpoints of various types and a Sources exception with one of these endpoints could prevent the entire list from being loaded and returned to the caller. Sources exceptions can be thrown for many reasons, one of them being the absence of data on the Sources side while the Notifications DB contains an ID that points to a Sources DB entry.

It makes complete sense to me to make sure a failure with one endpoint does not prevent the other endpoints from being loaded. Now, what do we do with the endpoint affected by the failure?
- Return inconsistent data? (current code in this PR)
- Exclude the endpoint from the returned list? (weird from a customer perspective)
- Return something that shows the endpoint exists but could not be loaded properly?

Reply:
We should return something to let the user knows that there was a problem loading the endpoint - but I'm not sure how to handle that - maybe a different http code and a custom header with the error? or  a different response that informs the error and wraps the endpoint list ? i.e. `{ errors: [ {} ]}, endpoints: [...]}`

But i agree that we should not return inconsistent data.

As a side note - Is it OK to continue retrieving all the passwords? (I'm guessing there could be security concerns here).

Wondering if secrets has (or has plans to) a UI to securely view the secrets - if that's the case we could simply link to secrets UI or similar.